src/sliceDataStorage.h

```diff
@@ -144,15 +183,27 @@ class SliceLayer
 };
 
 /******************/
+
+
+
+
 class SupportLayer
 {
 public:
-    Polygons supportAreas; //!< normal support areas
-    Polygons support_bottom; //!< Piece of support below the support and above the model. This must not overlap with supportAreas or support_roof.
-    Polygons support_roof; //!< Piece of support above the support and below the model. This must not overlap with supportAreas or support_bottom.
+    std::vector<SupportInfillPart> support_infill_parts;  //!< a list of support infill parts
+    Polygons support_bottom; //!< Piece of support below the support and above the model. This must not overlap with the global support areas (union of all support infill part outlines) or support_roof.
```

Comment:
Minor point:
you could also describe this as "This must not overlap with any of the support_infill_parts or support_roof."
[not blocking]